sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -196,6 +210,58 @@ dif_rstmgr_result_t dif_rstmgr_reset_info_get(
 DIF_WARN_UNUSED_RESULT
 dif_rstmgr_result_t dif_rstmgr_reset_info_clear(const dif_rstmgr_t *handle);
 
+/**
+ * Enables or disables the alert crash dump capture.
+ *
+ * When enabled, will capture the alert crash dump prior to a triggered reset.
+ *
+ * @param handle A Reset Manager handle.
+ * @param state The new toggle state for the crash dump capture.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_alert_info_set_enabled(
+    const dif_rstmgr_t *handle, dif_rstmgr_toggle_t state);
+
+/**
+ * Retrieves the size of the alert info crash dump.
+ *
+ * The corresponding registers are always-on and are expected to retain
+ * the information after all type of resets except Power-On-Reset (POR).
+ *
+ * @param handle A Reset Manager handle.
+ * @param count The size of alert info crash dump in number of
+ *              `dif_rstmgr_alert_dump_value_t`.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_get_size(
+    const dif_rstmgr_t *handle, size_t *count);
+
+/**
+ * Alert info crash dump 32-bit value.
+ */
+typedef uint32_t dif_rstmgr_alert_info_dump_value_t;
+
+/**
+ * Reads a single alert info crash dump `value` at `index`.
+ *
+ * Alert info crash dump consists of one or more
+ * `dif_rstmgr_alert_info_dump_value_t`. The total count can be found through
+ * using the `dif_rstmgr_alert_info_dump_get_size`.
+ *
+ * The corresponding registers are always-on and are expected to retain
+ * the information after all type of resets except Power-On-Reset (POR).
+ *
+ * @param handle A Reset Manager handle.
+ * @param count The size of alert info crash dump.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_rstmgr_result_t dif_rstmgr_alert_info_dump_read(
+    const dif_rstmgr_t *handle, size_t index,
+    dif_rstmgr_alert_info_dump_value_t *value);
```

Comment:
It seems there's a lot of suggestions for this API, and the like. 

I'd like to avoid a stateful cursor-based API, so I think the solution might just be to go with a single function that reads all the dump values into a buffer, which also says how many entries are in there when it's complete. I think we can bound the size of the buffer required because we know how many bits there are in the INDEX field used for accessing them. This feels nicer than a stateful API or re-reading the size field on every access to make sure the index we chose was in range.

Reply:
Going with @alphan [solution](https://github.com/lowRISC/opentitan/pull/3867/commits/b7d4b68f4a97a0c445df3881b948c1cc2c59e7a9#r508598310).